src/examples/fake_device_demo.rb

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env ruby
+
+$LOAD_PATH.unshift(File.expand_path('../../lib', __FILE__))
+
+require "storage/fake_probing.rb"
+require "storage/fake_device_factory.rb"
+
+FILENAME = "fake-devicegraphs"
+
+fake_probing = Yast::Storage::FakeProbing.new
+devicegraph = fake_probing.devicegraph
+factory = Yast::Storage::FakeDeviceFactory.new(devicegraph)
+factory.load_yaml_file("#{FILENAME}.yml")
+
+fake_probing.to_probed
+puts("Disks:")
+probed = Yast::Storage::StorageManager.instance.probed
+fake_probing.dump_disks(probed)
```

Comment:
If the main functionality of `FakeProbing`is creating the initial instance of libstorage to avoid the real probing and its secondary functionality is to set `StorageManager.probed` to a fake one. I think a code like this is more obvious about those intentions:

```ruby
using FakedStorageManager # If you don't want to "pollute" StoreManager itself

StorageManager::create_fake_instance
StorageManager.instance.probed = a_method_returning_the_devicegraph
```

Than this other one
```ruby
fake_probing = FakeProbing.new # This initializes StorageManager.instance as a non-obvious side effect
a_method_populating_a_given_devicegraph(fake_probing.devicegraph)
fake_probing.to_probed # A message sent to fake_probing that causes a change in the state of StorageManager.instance
```
Second proposal is sending messages to A and expecting the status of B to change. First proposal sends message directly to B, even if it implies extending its API (something that can be done cleanly only in the test code with a refinement).

Another side note (not 100% relevant to the current discussion), having `dump_disks` in FakeProbing is something that doesn't fit with the main or secondary responsibility (avoiding the probe and setting `probed`).